contracts/0.8.9/SepoliaDepositAdapter.sol

```diff
@@ -4,24 +4,77 @@
 /* See contracts/COMPILERS.md */
 pragma solidity 0.8.9;
 
+import "@openzeppelin/contracts-v4.4/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-v4.4/access/Ownable.sol";
 
-contract SepoliaDepositAdapter {
-
-    uint public constant TEST_VALUE = 16;
-    address public immutable depositContract;
+// Sepolia deposit contract variant of the source code https://github.com/protolambda/testnet-dep-contract/blob/master/deposit_contract.sol
+interface ISepoliaDepositContract is IERC20 {
 
     function deposit(
         bytes calldata pubkey,
         bytes calldata withdrawal_credentials,
         bytes calldata signature,
         bytes32 deposit_data_root
-    ) external payable {
-    }
+    ) external payable;
+
+    function get_deposit_root() external view returns (bytes32);
+
+    function get_deposit_count() external view returns (bytes memory);
+
+    function name() external view returns (string memory);
+}
+
+contract SepoliaDepositAdapter is Ownable {
```

Comment:
I would put a piece of comment for future readers: why we need this adapter and what is the idea behind it :eyes: